compiler/rustc_error_codes/src/error_codes/E0212.md

```diff
@@ -0,0 +1,52 @@
+An associated type cannot be extracted from
```

Comment:
I find this description confusing, I had to read the code to see what it meant. I know you took it from the error itself, but could we simplify somehow? Maybe "cannot use the associated type of a trait with uninferred generic parameters"? Ideally you'd also update the error itself, not just the long explanation.

Reply:
I agree and `"cannot use the associated type of a trait with uninferred generic parameters"` seems good. I'll try to update the error itself and update the PR description.